batch/batch/server/server.py

```diff
@@ -480,6 +490,7 @@ def is_successful(self):
 
     async def mark_unscheduled(self):
         if self._pod_name:
+            await _delete_pod_by_name(self._pod_name)
```

Comment:
I think you should move the db update to the function _delete_pod_by_name

Reply:
I'd have to sometimes do update_record and sometimes update_with_log_ec, I think it's better to keep them separate for now.